pkg/core/fs.go

```diff
@@ -50,7 +52,12 @@ type MutableFS struct {
 }
 
 // NewReadOnlyFS creates a new instance of the datamon filesystem.
+// todo: control flow regarding whether to cache pivots on this function.  ideas:
```

Comment:
this api design quandry is the only additional `todo` incurred with this diff .. would appreciate opinions regarding this code design decision in particular

Reply:
The property for what kind of bundle it is makes sense. Is it pre fetched, or streamed.

Reply:
deferring this for now..

Reply:
i've gone with the option of consolidating the `NewReadOnlyFS` functions according to `BundleOption`s